test/integration/slack_test.exs

```diff
@@ -61,4 +61,58 @@ defmodule Integration.SlackTest do
     message = send_message("@#{@bot}: operable:echo blah", private_channel)
     assert_response "blah", [after: message], private_channel
   end
+
+  test "redirecting from a private channel", %{user: user} do
+    user |> with_permission("operable:echo")
+    private_channel = "group_ci_bot_testing"
+    marker = send_message("Redirect test starts here")
```

Comment:
Reading through other tests in this module, it may be useful to introduce keyword arguments for the originating room... that way you could have code like `send_message("blah blah blah", from: "my_channel")`

That could be a follow-on, change, though, as it would end up affecting other tests, too.